openedx/core/djangoapps/lang_pref/tests/test_middleware.py

```diff
@@ -100,3 +100,30 @@ def test_delete_user_lang_preference_not_supported_by_system(self):
         set_user_preference(self.user, LANGUAGE_KEY, 'eo')
         self.middleware.process_request(self.request)
         self.assertEqual(get_user_preference(self.request.user, LANGUAGE_KEY), None)
+
+    @mock.patch(
+        'openedx.core.djangoapps.lang_pref.middleware.released_languages',
+        mock.Mock(return_value=[('eu-es', 'euskara (Espainia)'), ('en', 'english')])
+    )
+    def test_supported_browser_language_prefix_in_session(self):
+        """
+        test: browser language should be set in user session if it's prefix is supported by system for 
+        unathenticated users 
+        """
+        self.request.META['HTTP_ACCEPT_LANGUAGE'] = 'eu;q=1.0'
+        self.request.user = self.anonymous_user
+        self.middleware.process_request(self.request)
+        self.assertEqual(self.request.session.get(LANGUAGE_SESSION_KEY), 'eu-es')  #pylint: disable=no-member
+    
+    @mock.patch(
+        'openedx.core.djangoapps.lang_pref.middleware.released_languages',
+        mock.Mock(return_value=[('en', 'english')])
+    )
+    def test_browser_language_prefix_not_be_in_session(self):
+        """
+        test: browser language should not be set in user session if it's prefix is not supported by system.
+        """
+        self.request.META['HTTP_ACCEPT_LANGUAGE']='eu;q=1.0'
+        self.request.user = self.anonymous_user
+        self.middleware.process_request(self.request)
+        self.assertNotEqual(self.request.session.get(LANGUAGE_SESSION_KEY), 'eu-es')   # pylint: disable=no-member
```

Comment:
Oh! Yes, you're right. 
I have been working on a different localization library, with (of course) a different set of pitfalls and applied them here. Thank you for the explanation, and I am going to approve your PR now (and go read more about lang attributes).
Thank you again! Good stuff!